### ISSUE TITLE
bug: `enum` dynamic singleton methods not inherited the second time

### ISSUE THREAD

**Issue by thefliik** [neo4jrb/neo4j#1412]
This is a resubmit of #1410, this time I've tested on a new, clean neo4j test-project.
Given this model:
```
class Role
  include Neo4j::ActiveNode

  enum template: [:sunday, :monday]

  class Template < Role
    class Current < Template
    end
  end
end
```

## Expected Behavior
```
Role.sunday
#=> #<QueryProxy  []>

Role::Template.sunday
#=> #<QueryProxy  []>

Role::Template::Current.sunday
#=> #<QueryProxy  []>
```

## Actual Behavior

```
Role.sunday
#=> #<QueryProxy  []>

Role::Template.sunday
#=> #<QueryProxy  []>

Role::Template::Current.sunday
#=> NoMethodError: undefined method `arity' for nil:NilClass
```

### Runtime information:

Neo4j database version: 3.2.1
`neo4j` gem version: 8.2.1
`neo4j-core` gem version: 7.2.3

**Comment by thefliik** [comment]
After some digging, I believe this is caused by issue #1413, as `Role::Template::Current.sunday` is defined dynamically as a scope [here](https://github.com/neo4jrb/neo4j/blob/131a5de51916a427460286aa6f6e2465c3d726b9/lib/neo4j/active_node/enum.rb#L19).

**Comment by cheerfulstoic** [comment]
Should be fixed be #1415 in 8.2.3
